tldw_Server_API/app/core/MCP_unified/protocol.py

```
@@ -1340,10 +1340,17 @@ async def process_request(
                     result = await handler(request.params or {}, context)
                     span.set_attribute("mcp.status", "success")
                 except _MCP_PROTOCOL_NONCRITICAL_EXCEPTIONS as _span_e:
+                    sanitized = self._sanitize_exception_for_telemetry(_span_e)
                     span.set_attribute("mcp.status", "failure")
-                    span.set_attribute("mcp.error_type", _span_e.__class__.__name__)
-                    span.set_attribute("mcp.error_message", str(_span_e)[:200])
-                    raise
+                    span.set_attribute("mcp.error_type", sanitized.__class__.__name__)
+                    span.set_attribute("mcp.error_message", str(sanitized)[:200])
+                    raise sanitized from None
+                except Exception as _span_e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)